src/widgets/gitClone.ts

```diff
@@ -84,9 +86,41 @@ export class GitClone extends Widget {
   private makeApiCall(cloneUrl: string) {
     this.gitApi
       .clone(this.fileBrowser.model.path, cloneUrl)
-      .then(response => {
-        if (response.code !== 0) {
-          this.showErrorDialog(response.message);
+      .then(async response => {
+        let retry = false;
+        while (response.code !== 0) {
+          if (
+            response.code === 128 &&
+            (response.message.indexOf('could not read Username') >= 0 ||
+              response.message.indexOf('Invalid username or password') >= 0)
```

Comment:
Good idea. I put the constant string in `src/git.ts` and then just import it in relevant places.

I decided to use `map` instead of loop to check if response message contains any of the error strings.